boards/shared/Test/Src/Test_ErrorTable.cpp

```diff
@@ -754,3 +758,450 @@ TEST_F(
         ASSERT_TRUE(App_SharedError_IsBoardInList(&board_list, board));
     }
 }
+
+TEST_F(SharedErrorTableTest, process_bms_non_critical_errors)
+{
+    // A list of BMS non-critical error IDs. This must be maintained manually as
+    // errors are added to/removed from the DBC!
+    std::vector<enum ErrorId> bms_non_critical_error_ids = {
```

Comment:
If there's a way to _not_ have the user manually maintain this list, that would be helpful. But as of now I can't think of anything.

Reply:
This applies to all the boards

Reply:
One option could be to generate 
```
#define BMS_NON_CRITICAL_ERROR_IDS \
BMS_NON_CRITICAL_STACK_WATERMARK_ABOVE_THRESHOLD_TASK1HZ, \
        BMS_NON_CRITICAL_STACK_WATERMARK_ABOVE_THRESHOLD_TASK1KHZ, \
        BMS_NON_CRITICAL_STACK_WATERMARK_ABOVE_THRESHOLD_TASKCANRX, \
        BMS_NON_CRITICAL_STACK_WATERMARK_ABOVE_THRESHOLD_TASKCANTX, \
        BMS_NON_CRITICAL_WATCHDOG_TIMEOUT. \
```

And stick it into `enum ErrorId`?

```
#include <SomeAutoGenHeader.h>
enum ErrorId
{
    BMS_NON_CRITICAL_ERROR_IDS,
    DCM_NON_CRITICAL_ERROR_IDS,
    ...
    BMS_CRITICAL_ERROR_IDS,
    DCM_CRITICAL_ERROR_IDS,
    ....
    NUM_ERROR_IDS,
}
```

And this test case would be rewritten as, where the user doesn't need to deal with the mental overhead of modifying this error ID list every time a BMS non-critical error is added to the DBC
```
TEST_F(SharedErrorTableTest, process_bms_non_critical_errors)
{
    std::vector<enum ErrorId> bms_non_critical_error_ids = {
       BMS_NON_CRITICAL_ERROR_IDS
    };

    TestRoutineForSetErrorsFromCanMsg(
        BMS, bms_non_critical_error_ids,
        CANMSGS_BMS_NON_CRITICAL_ERRORS_FRAME_ID,
        CANMSGS_BMS_NON_CRITICAL_ERRORS_LENGTH,
        App_CanMsgs_bms_non_critical_errors_pack,
        App_SharedErrorTable_GetBoardsWithNonCriticalErrors,
        App_SharedErrorTable_GetAllNonCriticalErrors);
}
````

Reply:
Honestly I'd say generating the enum is fine, and I don't think we need to check the code in?

Reply:
Sounds good let me go add in a script for this !

Reply:
Just to double check, this PR is still blocked on this script?

Reply:
Yup still working on it!

Reply:
Finally got around to fixing this!